akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/IllegalHeadersIntegrationSpec.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server.directives
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.headers.{ Accept, RawHeader }
+import akka.http.scaladsl.server.IntegrationRoutingSpec
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+/**
+ * Has to excercise the entire stack, tgus an IntegrationRoutingSpec (not reproducable using just RouteTest).
+ */
+class IllegalHeadersIntegrationSpec extends IntegrationRoutingSpec {
+
+  "Illegal content type in request" should {
+    val route = extractRequest { req ⇒
+      complete(s"Accept:${req.header[Accept]}, byName:${req.headers.find(_.is("accept"))}")
+    }
+
+    // see: https://github.com/akka/akka-http/issues/1072
+    "not StackOverflow but be rejected properly" in {
+      val theIllegalHeader = RawHeader("Accept", "*/xml")
```

Comment:
Ah, didn't see that this was an IntegrationRoutingSpec (or that we actually had something like that), so that the headers are run through the whole infrastructure.